src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -43,9 +44,15 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnRSessionInteractive(object sender, EventArgs e) {
+        private void OnRSessionConnected(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
+                // Host signals 'connected' after R loop was set up (and R banner is printed).
+                // We want to complete printing synchorously here so output does not come
+                // after user started typing in the REPL. But don't wait forever.
+                Task.Run(async () => {
+                    await ReplInitComplete();
+                    await PrintBrokerInformationAsync(reportTelemetry: true);
+                }).Wait(10000);
```

Comment:
I looked at R and our host code and the banner is printed in `setupRLoop` which we call (and then offer MRO), then host calls 'connected'. So by the 'connected' time the banner should be in the REPL queue. However, releasing control here will allow other interactions to happen - such as initlialization process and possible even user input.